client/components/Profiles/ChangeProfilesButton.tsx

```diff
@@ -0,0 +1,103 @@
+import React, { useState } from 'react'
+import { useDataContext } from '@/context/globalContext';
+import { ProfileType } from '@/types';
+import ProfileButton from './ProfileButton';
+import { SafeAreaView } from 'react-native-safe-area-context';
+
+const ChangeProfilesButton = () => {
+    const { profiles, selectedProfile, setSelectedProfile } = useDataContext();
+    const [modalVisible, setModalVisible] = useState(false);
+    
+    const handleProfilePress = (profile: ProfileType) => {
+        setSelectedProfile(profile);
+        setModalVisible(false);
+      };
+
+  return (
+    <SafeAreaView>
+      <Pressable 
+        style={styles.circleButton} 
+        onPress={() => setModalVisible(true)}
+      > 
+      {selectedProfile ? (
+          selectedProfile.picture ? (
+            <Image source={{ uri: selectedProfile.picture }} style={styles.profileImage} />
+          ) : (
+            <Text style={styles.profileInitial}>
+              {selectedProfile.name.charAt(0).toUpperCase()}
+            </Text>
+          )
+        ) : (
+          <Text style={styles.circleButtonText}>+</Text>
+        )}
+      </Pressable>
+      <Modal
+        transparent={true}
+        visible={modalVisible}
+        animationType="fade"
+        onRequestClose={() => setModalVisible(false)}
+      >
+        <View style={styles.modalContainer}>
+          <FlatList
+            data={profiles}
+            keyExtractor={(item) => item.id.toString()}
+            renderItem={({ item }) => (
+              <ProfileButton
+                    profileName={item.name}
+                    buttonStyle={styles.profileButton}
+                    textStyle={styles.profileButtonText}
+                    onPress={() => handleProfilePress(item)} 
+                    route={'/homeScreen'}             
+                />
+            )}
+          />
+        </View>
+      </Modal>
+    </SafeAreaView>
+  )
+}
+
+const styles = StyleSheet.create({
+    circleButton: {
+        backgroundColor: '#28a745',
+        borderRadius: 50,
+        width: 50,
+        height: 50,
+        justifyContent: 'center',
+        alignItems: 'center',
+      },
+      profileImage: {
+        width: 50,
+        height: 50,
+        borderRadius: 50,
+      },
+      profileInitial: {
+        color: '#fff',
+        fontSize: 24,
+      },
+      circleButtonText: {
+        color: '#fff',
+        fontSize: 24,
+      },
+      modalContainer: {
+        flex: 1,
+        justifyContent: 'center',
+        alignItems: 'center',
+        backgroundColor: 'rgba(0,0,0,0.5)',
+      },
+      profileButton: {
+        backgroundColor: '#fff',
+        padding: 15,
+        marginVertical: 5,
+        width: 200,
+        borderRadius: 5,
+      },
+      profileButtonText: {
+        fontSize: 18,
+        fontWeight: 'bold',
+        color: '#000',
+      },
+    });
+
+export default ChangeProfilesButton;
```

Comment:
this is nitpicking because I couldn't find anything else, but it's good to know for future reference. All files should have an empty line at the end. Thats why GH marks it like this. Just FYI, you can leave it.